grpc.go

```diff
@@ -867,6 +867,30 @@ func (a *agentGRPC) StatsContainer(ctx context.Context, req *pb.StatsContainerRe
 
 }
 
+func (a *agentGRPC) PauseContainer(ctx context.Context, req *pb.PauseContainerRequest) (*gpb.Empty, error) {
+	c, err := a.sandbox.getContainer(req.ContainerId)
+	if err != nil {
+		return emptyResp, err
+	}
+
+	a.sandbox.Lock()
+	defer a.sandbox.Unlock()
```

Comment:
why are we using locks here ? I don't get the use case for that

Reply:
@devimc Hi, please check this.